torch/overrides.py

```diff
@@ -1726,7 +1714,7 @@ class TorchFunctionModeMeta(type):
     """
     def __new__(metacls, name, bases, dct):
         if '__init__' in dct:
-            dct['__init__'] = _wrap_init(dct['__init__'])
+            dct['__init__'] = _wrap_init(dct['__init__'], "TorchFunctionMode", "torch_function")
```

Comment:
Don't want to generalize this for both function and dispatch? ;)

Reply:
oh this might just be dead code now?

Reply:
This ended up not actually being dead code. I definitely got confused on that too

> Don't want to generalize this for both function and dispatch? ;)

Sorry I think I'm parsing this weirdly--are you saying we shouldn't generalize the `_wrap_init`?

Reply:
My main question is if you have a dispatch/function generic metaclass for modes, then why do we need TorchFunctionModeMeta

Reply:
Oh I see--ModeMeta is actually the dead code, not this (but also deleting that, I missed it)

I originally had that when I thought I could make this generic but that piece of code actually didn't work so I started passing everything as strings to avoid circular dependencies. At minimum, I'll put these strings into a little dataclass with some comments so it's clearer why they're being used (again mostly to avoid circular dependencies)

Reply:
Added the dataclass with a comment on the parent class but leaving this open in case we want more discussion